src/core.js

```diff
@@ -425,6 +425,16 @@ jQuery.extend({
 	support: support
 });
 
+// jsHint would error on this code due to the Symbol not being defined in ES5.
+// Defining this global in .jshintrc would create a danger of using the global
+// unguarded in another place, it seems safer to just disable jsHint for these
+// three lines.
+/* jshint ignore: start */
+if ( typeof Symbol === "function" && Symbol.iterator ) {
```

Comment:
This is tough because I agree with both @ljharb and @mzgol on this point. On one hand I'm not convinced that half-shimming features is a good idea; on the other, I think as long as this allows for the widest net possible, it's totally ok. It's hard to say how or if it will break in the wild, but I don't think you're going to know until you try. It's a bold move and I like it.